server/controllers/stock/core.js

```diff
@@ -78,8 +85,9 @@ function getLots(sqlQuery, parameters, finalClauseParameter) {
 
   filters.dateFrom('dateFrom', 'date', 'm');
   filters.dateTo('dateTo', 'date', 'm');
+  filters.setGroup(finalClause || '');
```

Comment:
Wow, this works but it is confusing.  Could you put a comment above this line telling future programmers that `FilterParser` sets the grouping to an empty string when it doesn't exist?  I had to look it up ([filter.js#L57](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/lib/filter.js#L57).)